### ISSUE TITLE
Where is the skip connection part?

### ISSUE THREAD

**Issue by YNX940214** [Yijunmaverick/DeepJointFilter#3]
I have read the code carefully but can't find the skip connection part.
And i have several questions:

1> In the code , the input patch size is 33x33, but 32x32 in the paper?

2> In the paper with skip connection, is the output patch size the same with input, that is 33x33 ,different from the matlab implemention 21x21? 

3>And it is strange: I have re-implement the no-skip connection version with pytorch and reach the RMSE of 5.9 (lower than 6.2 in the paper) , but when i add the skip connection with the training parameter unchanged (I set these parameters with the running data of the matlab code), the RMSE got worse. What should I do? 

4>How can i further improve the RMSE? 

Thanks.

**Comment by Yijunmaverick** [comment]
Hi,

1) See `GoTraining.m` where I just set different labels for different training settings.
2) The output size is also 21x21. I remove the boundary pixels (33-12)x(33-12) of the input.
3) I do not try Pytorch yet but it usually needs some engineering effort to get a new implementation work. I can hardly comment on your case.
4) There are already a few papers with better performance than mine. You can check them and hopefully get some ideas, e.g., [DGDE](http://www4.comp.polyu.edu.hk/~cslzhang/paper/DGDE_CVPR17.pdf).

Thanks.

**Comment by YNX940214** [comment]
Thank your for answering.
I still have one question: the matlab code doesn't have the skip connection ? Is there any differences except for the padding removal and learning rate from 1e-4 to 1e-3 betweent the two versions?
@Yijunmaverick

**Comment by Yijunmaverick** [comment]
I do not explicitly add a skip link in the network. Please see `GoTraining.m` where I just change label from `GT depth` to `GT depth - Input depth` for the residual learning.

My experience is the that you still need to spend efforts to get the residual version work. It is not as easy as simply adding the skip link and training the network without tuning parameters.

**Comment by YNX940214** [comment]
Thanks Doctor.
